internal/kafka/internal/config/dynamic_scaling_config.go

```diff
@@ -51,25 +55,57 @@ func (c *DynamicScalingConfig) validate() error {
 	return nil
 }
 
-type InstanceTypeDynamicScalingConfig struct {
-	ComputeNodesConfig *DynamicScalingComputeNodesConfig `yaml:"compute_nodes_config"`
+type ComputeNodesAutoscalingConfig struct {
+	MaxComputeNodes int `yaml:"max_compute_nodes" validate:"gt=0,gtefield=MinComputeNodes"`
+	MinComputeNodes int `yaml:"min_compute_nodes" validate:"gt=0"`
 }
 
-func (c *InstanceTypeDynamicScalingConfig) validate() error {
-	if c.ComputeNodesConfig == nil {
-		return fmt.Errorf("compute_nodes_config is mandatory")
-	}
+type ComputeMachineConfig struct {
+	ComputeMachineType      string                         `yaml:"compute_machine_type" validate:"required"`
+	ComputeNodesAutoscaling *ComputeNodesAutoscalingConfig `yaml:"compute_node_autoscaling" validate:"required"`
+}
 
-	return c.ComputeNodesConfig.validate()
+type ComputeMachinesConfig struct {
+	ClusterWideWorkload          *ComputeMachineConfig           `yaml:"cluster_wide_workload" validate:"required"`
+	KafkaWorkloadPerInstanceType map[string]ComputeMachineConfig `yaml:"kafka_workload_per_instance_type" validate:"required"`
 }
 
-type DynamicScalingComputeNodesConfig struct {
-	MaxComputeNodes int `yaml:"max_compute_nodes"`
+func (c *ComputeMachinesConfig) GetConfigForInstanceType(instanceTypeID string) (ComputeMachineConfig, bool) {
```

Comment:
Probably better renamed to `GetKafkaWorkloadConfigForInstanceType`

Reply:
Sounds good. Will update